--- /Users/nicktrienens/Projects/SwiftUIPagedScrolling/Sources/SwiftUIPagedScrolling/SwiftUIPagedScrolling.swift
+++ /Users/nicktrienens/Projects/SwiftUIPagedScrolling/Sources/SwiftUIPagedScrolling/SwiftUIPagedScrolling.swift
@@ -2,6 +2,12 @@
 
+public enum PagerGesturePriority {
+    case standard
+    case high
+    case simultaneous
+}
+
 public struct SwiftUIPagedScrolling<Data: RandomAccessCollection, ID: Hashable, Content: View>: View {
     private let data: Data
     private let id: KeyPath<Data.Element, ID>
@@ -22,12 +28,6 @@
         case vertical
     }
 
-    public enum PagerGesturePriority {
-        case standard
-        case high
-        case simultaneous
-    }
-
     public init(
         data: Data,
         id: KeyPath<Data.Element, ID>,
@@ -62,7 +62,6 @@
             let shouldDisableScrolling = gesturePriority == .simultaneous
             let disableScrolling = shouldDisableScrolling && isDragging && ((orientation == .horizontal && dragDirection == .horizontal) || (orientation == .vertical && dragDirection == .vertical))
 
-            let dragGesture = DragGesture(minimumDistance: 15, coordinateSpace: .local)
-
             ZStack(alignment: isHorizontal ? .leading : .top) {
                 if !data.isEmpty {
@@ -79,8 +78,8 @@
                 x: isHorizontal ? -CGFloat(isDragging ? gestureStartIndex : currentIndex) * totalDimension + offset : 0,
                 y: isHorizontal ? 0 : -CGFloat(isDragging ? gestureStartIndex : currentIndex) * totalDimension + offset
             )
-            .simultaneousGesture(
-                DragGesture(minimumDistance: 15, coordinateSpace: .local)
+            .applyPagerGesture(
+                gesture: DragGesture(minimumDistance: 15, coordinateSpace: .local)
                     .onChanged { value in
                         if !isDragging {
                             isDragging = true
@@ -155,7 +154,8 @@
                         }
                         dragDirection = nil
-                    }
+                    },
+                priority: gesturePriority
             )
         }
         .clipped()
     }
@@ -165,3 +165,16 @@
     }
 }
+
+private extension View {
+    @ViewBuilder
+    func applyPagerGesture<G: Gesture>(gesture: G, priority: PagerGesturePriority) -> some View {
+        switch priority {
+        case .standard:
+            self.gesture(gesture)
+        case .high:
+            self.highPriorityGesture(gesture)
+        case .simultaneous:
+            self.simultaneousGesture(gesture)
+        }
+    }
+}
